keys/keys.go

```diff
@@ -261,10 +261,10 @@ func ValidateRangeMetaKey(key proto.Key) error {
 	prefix, body := proto.Key(key[:len(Meta1Prefix)]), proto.Key(key[len(Meta1Prefix):])
 
 	if prefix.Equal(Meta2Prefix) {
```

Comment:
you're pretty much doing the same thing for Meta1 or Meta2 keys. Consider merging this here with the one below.